singularity-eos/eos/modifiers/eos_unitsystem.hpp

```diff
@@ -269,6 +269,13 @@ class UnitSystem : public EosBase<UnitSystem<T>> {
     printf("Units = %e %e %e %e\n", rho_unit_, sie_unit_, temp_unit_, press_unit_);
   }
 
+  PORTABLE_FORCEINLINE_FUNCTION
+  bool IsModified() const { return true; }
+  PORTABLE_FORCEINLINE_FUNCTION
+  T UnmodifyOnce() { return t_; }
+  PORTABLE_FORCEINLINE_FUNCTION
+  auto GetUnmodifiedObject() { return t_.GetUnmodifiedObject(); }
```

Comment:
Unfortunately, these are copy-pasted into each modifier definition. It's maybe possible to add them all with a macro or some other trick, although I don't know how worthwhile that is.

Reply:
Two tier CRTP? There would be the standard base class and then a modifier base class that inherits from the standard base class and provides these functions. I played around with it a bit a while back and it is doable.
https://stackoverflow.com/a/18174442/3491638

Reply:
Hmm... that could work. If we add more modifier-specific stuff, let's consider it.

Reply:
For consideration: mixins fit this pattern well, tho (from experience) I can say using mixins and crtp can be give tongue-twisty code. https://www.fluentcpp.com/2017/12/12/mixin-classes-yang-crtp/

Reply:
If @jhp-lanl wants to make an PR, I'd be interested to see how tongue-twisty it is. May be worth it, though the current duplication overhead is small and not likely to be modified often.

Reply:
@mauneyc-LANL I was looking at mixins as well a little bit ago but I thought that they weren't as elegant with our `Variant` class. We define a closed set of functions that are available for all types, whereas I feel a mixin is about expanding the functionality. If we were using class inheritance for our type erasure, maybe a mixin would be the preferable approach, but with the Variant, I lean towards CRTP since the end user doesn't have to change anything.

However, I wonder if a mixin could enable functionality like this?
```c++
EOS my_modified_eos =  Shifted<Scaled<IdealGas(...)>>>;
EOS once_unmodified = UnmodifyOnce(my_modified_eos);
EOS unmodified_eos = UnmodifyAll(my_modified_eos);
```